word_cloud_generator/word_cloud_generator.py

```diff
@@ -0,0 +1,307 @@
+    """
+    fields = issue.get("fields", None)
+    if not fields:
+        return False
+
+    assignee = fields.get("assignee", None)
+    if not assignee:
+        return False
+
+    issue_assigned = assignee.get("displayName", None)
+    assign_check = issue_filter.assigned
+    if (not issue_assigned or issue_assigned != assign_check) and assign_check:
+        return False
+
+    if issue_date > datetime.strptime(issue_filter.start_date, "%Y-%m-%d"):
+        return False
+    return True
+
+
+def generate_word_cloud(
+    issues_contents, issue_filter, word_cloud_output_location, **kwargs
+):
+    """
+    Function to generate and save a word cloud
+    :param issues_contents: The summary of every valid issue (list)
+    :param issue_filter: Dict of filters to check the issues against (dict)
+    :param word_cloud_output_location: The output location for the word cloud to be saved to
+    :param kwargs: A set of kwargs to pass to WordCloud
+    - width
+    - height
+    - min_font_size
+    - max_words
+    """
+    matches = re.findall(r"((\w+([.'](?![ \n']))*[-_]*)+)", issues_contents)
+    # Regex to find all words and include words joined with certain characters, while not
+    # allowing certain characters to exist at the start or end of the word, such as dots.
+    if matches:
+        issues_contents = " ".join(list(list(zip(*matches))[0]))
+    issues_contents = filter_word_cloud(issue_filter, issues_contents)
+    word_cloud = WordCloud(
+        width=kwargs["width"],
+        height=kwargs["height"],
+        min_font_size=kwargs["min_font_size"],
+        max_words=kwargs["max_words"],
+        background_color="white",
+        collocations=False,
+        regexp=r"\w*\S*",
+    )
+
+    word_cloud.generate(issues_contents)
+
+    word_cloud.to_file(word_cloud_output_location)
+
+
+def filter_word_cloud(issue_filter, issues_contents):
+    """
+    Function to filter the contents of the word cloud to or against certain strings
+    :param issues_contents: The summary of every valid issue (list)
+    :param issue_filter: Dict of filters to check the issues against (dict)
+    :returns: The filtered issues contents
+    """
+    if issue_filter.filter_not:
+        issues_contents = re.sub(
+            issue_filter.filter_not.lower(), "", issues_contents, flags=re.I
+        )
+    if issue_filter.filter_for:
+        issues_contents = " ".join(
+            re.findall(
+                issue_filter.filter_for.lower(),
+                issues_contents,
+                flags=re.IGNORECASE,
+            )
+        )
+
+    return issues_contents
+
+
+def word_cloud_generator():
+    """
+    Function to take arguments, generate the output location and run the
+    functions to the data for the word cloud and generate it
+    """
+    args = parse_args(argv[1:])
+    host = "https://stfc.atlassian.net"
+    username = args.username
+    password = args.password
+
+    issue_filter = from_user_inputs(**args.__dict__)
```

Comment:
I think you can use `vars()` here. That way you aren't using a private method.
```suggestion
    issue_filter = from_user_inputs(vars(args))
```